File: frontend/src/types/audit.ts
export interface AuditLog {
    id: string;
    actor_id: string | null;
    action: string;
    entity_type: string;
    entity_id: string;
    metadata: Record<string, unknown> | null;
    created_at: string;
}

export interface AuditLogListResponse {
    events: AuditLog[];
    page: number;
    page_size: number;
}

File: app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class AuditLogResponse(BaseModel):
    id: str
    actor_id: str | None
    action: str
    entity_type: str
    entity_id: str
    metadata: dict | None = Field(validation_alias="details")
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class AuditLogListResponse(BaseModel):
    events: list[AuditLogResponse]
    page: int
    page_size: int
